tough/src/transport.rs

```diff
@@ -156,6 +157,39 @@ impl Transport for FilesystemTransport {
         })?;
         Ok(Box::new(f))
     }
+
+    #[cfg(target_os = "windows")]
+    fn fetch(&self, url: Url) -> Result<Box<dyn Read + Send>, TransportError> {
+        if url.scheme() != "file" {
+            return Err(TransportError::new(
+                TransportErrorKind::UnsupportedUrlScheme,
+                url,
+            ));
+        }
+
+        // If the path returned from url.path() begins w/ "/" , we remove this "/" so windows doesn't complain
+        // This is dirty, but the only way i found to do it ＞﹏＜ ; any edits are welcome.
```

Comment:
Can you explain what is going on here? Do you have an example of one of these URLs that have a w/ as its first path component? Although I don't understand the issue being fixed here yet, I wonder if there might be a crate for this kind of thing if it is encountered frequently.

More generally, we should try to reduce the amount of code that is inside the `cfg`  switch to only the parts that are different (rather than having duplicate code in both static branched). But I'd like to understand the issue better before suggesting specific changes.

Reply:
So, what happens here is that the url.path() function, when called with a path that begins with the "file://" protocol returned paths that looked like : "/c:/foo/bar" instead of the expected "c:/foo/bar". 
This broke some tests with windows complaining about invalid file paths. ; I wrote a hack that removes, if found the "/" in front of the returned address. 
It looks like the crate is having issues in this case under windows and there looks like there are some issues on their repo about these kinds of problems, but I'm not sure of when it'll be fixed, so in the meantime, I wrote this to get it working.

Reply:
As an example, when I passed to the url.path() function an url that looked like "file:///c:/foo/bar", it returned "/c:/foo/bar", which wasn't a valid Windows path.